From 2d2a55a0635e5b411dca20196a496b880d7ba80e Mon Sep 17 00:00:00 2001
From: yao zhang1 <yao.zhang1@amlogic.com>
Date: Wed, 8 Feb 2023 18:42:09 +0800
Subject: [PATCH 92/95] build: Fixed c3 mini defconfig built error [1/1]

PD#SWPL-106998

Problem:
c3 mini defconfig built error

Solution:
Fixed c3 mini defconfig built error

Verify:
C3 aw402

Change-Id: Idbbe53d8775f82bfff880ac3ce65b57574cf15da
Signed-off-by: yao zhang1 <yao.zhang1@amlogic.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac-meson8b.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-meson8b.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-meson8b.c
index ae7135f07ec2..491b57fe0626 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-meson8b.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-meson8b.c
@@ -583,6 +583,8 @@ static int meson8b_dwmac_probe(struct platform_device *pdev)
 	return ret;
 }
 
+#if IS_ENABLED(CONFIG_AMLOGIC_ETH_PRIVE)
+#ifdef CONFIG_PM_SLEEP
 static void meson8b_dwmac_shutdown(struct platform_device *pdev)
 {
 	struct net_device *ndev = platform_get_drvdata(pdev);
@@ -596,8 +598,6 @@ static void meson8b_dwmac_shutdown(struct platform_device *pdev)
 		ret = dwmac->data->suspend(dwmac);
 }
 
-#if IS_ENABLED(CONFIG_AMLOGIC_ETH_PRIVE)
-#ifdef CONFIG_PM_SLEEP
 static int dwmac_suspend(struct meson8b_dwmac *dwmac)
 {
 	pr_info("disable analog\n");
@@ -768,7 +768,11 @@ static struct platform_driver meson8b_dwmac_driver = {
 #else
 	.remove = stmmac_pltfr_remove,
 #endif
+#if IS_ENABLED(CONFIG_AMLOGIC_ETH_PRIVE)
+#ifdef CONFIG_PM_SLEEP
 	.shutdown = meson8b_dwmac_shutdown,
+#endif
+#endif
 	.driver = {
 		.name           = "meson8b-dwmac",
 #if IS_ENABLED(CONFIG_AMLOGIC_ETH_PRIVE)
-- 
2.25.1

